src/heliclockter/__init__.py

```
@@ -98,24 +73,17 @@ def __init__(  # pylint: disable=unused-argument
 
             self.assert_aware_datetime(self)
 
-    if PYDANTIC_V1_AVAILABLE or PYDANTIC_V2_AVAILABLE:
-
-        @classmethod
-        def __get_validators__(cls) -> Iterator[Callable[[Any], datetime_tz | None]]:
-            yield cls._validate
-
-        @classmethod
-        def _validate(cls: type[DateTimeTzT], v: Any) -> DateTimeTzT | None:
-            if v is None:
-                return None
-
-            dt = v if isinstance(v, _datetime.datetime) else parse_datetime(v)
-            return cls.from_datetime(dt)
-
-    if PYDANTIC_V2_AVAILABLE:
+    # We don't require pydantic as a dependency, but add validate logic if it exists.
+    try:
```

If I understand correctly this will silently fail if the user uses pydantic v1 on an older supported python version. they might not understand why the validation is not working in those cases, right?

Should we check this somehow?

---

Ah good point, we can check that here indeed. But we probobably also constrain pydantic to `pydantic>=2.0.0` in `pyproject.toml`, I will do that.